lib/commands/exam.js

```diff
@@ -299,6 +348,42 @@ module.exports = TestCommand.extend({
         init = true;
       }
     };
+
+    events['after-tests-complete'] = browserExitHandler;
+
+    events['disconnect'] = function() {
+      // when test runs without launcher (with --no-launch) process object is not avaialble.
+      // To prevent handling exiting browser browser disconnects from errors `disconnect` callback's needed to be registered.
+      if (typeof this.process === 'undefined' || this.process === null) {
+        browserFailureHandler.bind(this)();
+      }
+    }
+
+    events['testem:module-done-metadata'] = (details) => {
+      // Ensure module detail is available
+      if (typeof details === 'object' && details !== null) {
+        this.testemEvents.recordModuleMetaData(
+          {
+            name: details.name,
+            total: details.total,
+            runtime: details.runtime
+          }
+        );
+      }
+    };
+
+    return events;
+  },
+
+  /**
+   * Return an event object which enables load balancing.
+   * These event handlers will be registered on Testem's browserTestRunner socket instance
+   *
+   * @param {Object} commands
+   * @param {Object} testemEvents
+   */
+  _getLoadBalancingBrowserSocketEvents(commands, testemEvents) {
```

Comment:
instead of passing all of `this.commands`.. can we do 
```
this._getLoadBalancingBrowserSocketEvents({
  this.commands.get('loadBalance'),
  this.commands.get('replayExecution'),
  this.commands.get('writeExecutionFile')
}, this.testemEvents );
...
_getLoadBalancingBrowserSocketEvents({ isLoadBalance, isReplayExecution, isWriteExecutionFile} , testemEvents) {
```